libs/RobotKit/include/RobotController.h

```diff
@@ -0,0 +1,50 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _LEKA_OS_ROBOT_CONTROLLER_H_
+#define _LEKA_OS_ROBOT_CONTROLLER_H_
+
+#include "StateMachine.h"
+#include "interface/RobotController.h"
+#include "interface/drivers/Battery.h"
+
+namespace leka {
+
+template <typename sm_t = boost::sml::sm<system::robot::StateMachine>>
+class RobotController : public interface::RobotController
+{
+  public:
+	sm_t state_machine {static_cast<interface::RobotController &>(*this)};
+
+	explicit RobotController(interface::Battery &battery) : _battery(battery) {};
+
+	auto isCharging() -> bool final { return _battery.isCharging(); };
+
+	template <typename eventType>
+	void raise()
+	{
+		state_machine.process_event(eventType {});
+	};
+
+	void registerEvents()
+	{
+		// Initializations
+		// Setup callbacks for each events
+
+		auto on_charge_on = [this]() { raise<system::robot::sm::event::charge_did_start>(); };
+		_battery.onChargeDidStart(on_charge_on);
+
+		auto on_charge_off = [this]() { raise<system::robot::sm::event::charge_did_stop>(); };
+		_battery.onChargeDidStop(on_charge_off);
+
+		raise<system::robot::sm::event::setup_complete>();
```

Comment:
... ça! 

```suggestion
		auto on_charge_on = [this]() { raise(system::robot::sm::event::charge_did_start {}); };
		_battery.onChargeDidStart(on_charge_on);

		auto on_charge_off = [this]() { raise(system::robot::sm::event::charge_did_stop {}); };
		_battery.onChargeDidStop(on_charge_off);

		raise(system::robot::sm::event::setup_complete {});
```

l'argument est plus courant que d'utiliser les chevrons pour spécifier le type `<un::truc::très::long::et::très::illisible::>()`

Reply:
on peut même raccourcir avec un `using namespace`